### PR TITLE
Bug/fix postgresql queries

### LINKED ISSUE #97: Consumer unable to write a report to the db
When I upload an archive in the log I can see:
```
2020/02/18 15:53:47 Making connection to data storage, driver=postgres datasource=postgres://****:****@insights-results-aggregator-db.ccx-data-pipeline-ci.svc/aggregator?sslmode=disable
2020/02/18 15:53:47 Making connection to data storage, driver=postgres datasource=postgres://****:****@insights-results-aggregator-db.ccx-data-pipeline-ci.svc/aggregator?sslmode=disable
2020/02/18 15:53:47 Initializing HTTP server at :8080
2020/02/18 15:53:47 Starting HTTP server at :8080
2020/02/18 15:53:47 Consumer has been started, waiting for messages send to topic ccx.ocp.results
2020/02/18 15:54:04 Consumed message offset 640
2020/02/18 15:54:04 Results for organization 12383032 and cluster e8823fc9-337f-40b7-934c-2c90980fcd57
2020/02/18 15:54:04 Error writing report to database: pq: syntax error at or near "OR"
2020/02/18 15:54:04 Error processing message consumed from Kafka: pq: syntax error at or near "OR"
```

### PR DESCRIPTION
**Pull request by Sergey1011010**
# Description

fix postgresql queries and implemented logging sql queries for debugging

Fixes #97

## Type of change

Please delete options that are not relevant.

- Bug fix
- New feature
- This change requires a documentation update
- Refactor
- Unit tests

## Testing steps

`make test` & `make integration_tests`

### CODE REVIEW

File: storage/errors.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"fmt"
+)
+
+// ItemNotFoundError shows that item with id ItemID wasn't found in the storage
+type ItemNotFoundError struct {
+	ItemID interface{}
+}
+
+func (e *ItemNotFoundError) Error() string {
```
Comment (tisnik) [inline storage/errors.go]: please add docstring (it's straightforward there)

File: storage/sql_hooks.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const (
+	sqlHooksKeyQueryBeginTime sqlHooksKey = iota
+)
+
+// LogFormatterString is format string for sql queries logging
+// first arg is query string
+// second arg is params array
+const logFormatterString = "query `%+v` with params `%+v`"
+
+func (h *sqlHooks) Before(ctx context.Context, query string, args ...interface{}) (context.Context, error) {
```
Comment (tisnik) [inline storage/sql_hooks.go]: dtto - linter won't be happy about missing docstrings ;)

File: storage/sql_hooks.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	if err == nil {
+		h.SQLQueriesLogger.Printf(logFormatterString+"\n", query, string(jsonArgs))
+	} else {
+		h.SQLQueriesLogger.Printf(logFormatterString+"\n", query, args)
+	}
+
+	return context.WithValue(ctx, sqlHooksKeyQueryBeginTime, time.Now()), nil
+}
+
+func (h *sqlHooks) After(ctx context.Context, query string, args ...interface{}) (context.Context, error) {
```
Comment (tisnik) [inline storage/sql_hooks.go]: and there, please

File: storage/storage.go [hunk 4]
```diff
@@ -162,7 +202,7 @@ func (storage DBStorage) ListOfOrgs() ([]types.OrgID, error) {
 func (storage DBStorage) ListOfClustersForOrg(orgID types.OrgID) ([]types.ClusterName, error) {
 	clusters := []types.ClusterName{}
 
-	rows, err := storage.connection.Query("SELECT cluster FROM report WHERE org_id = ? ORDER BY cluster", orgID)
+	rows, err := storage.connection.Query("SELECT cluster FROM report WHERE org_id = $1 ORDER BY cluster", orgID)
```
Comment (tisnik) [inline storage/storage.go]: yeah, that's better. Bad me - dunno why I still use ? everywhere :)

File: storage/storage.go [hunk 5]
```diff
@@ -209,10 +254,19 @@ func (storage DBStorage) WriteReportForCluster(
...
+	switch storage.dbDriverType {
+	case DBDriverSQLite:
+		query = `INSERT OR REPLACE INTO report(org_id, cluster, report, reported_at, last_checked_at) 
+		 VALUES ($1, $2, $3, $4, $5)`
+	default:
+		query = `INSERT INTO report(org_id, cluster, report, reported_at, last_checked_at)
+		 VALUES ($1, $2, $3, $4, $5)
+		 ON CONFLICT (org_id, cluster) 
+		 DO UPDATE SET report = $3, reported_at = $4, last_checked_at = $5`
+	}
```
Comment (tisnik) [inline storage/storage.go]: unfortunately UPSERT is not defined+supported in SQL standard ;/ welcome to 2020

Comment (tisnik) [reply storage/storage.go]: I mean - your solution is correct ATM (unless we switch to Squirel)

File: storage/errors.go (hunk 1 above)
Comment (Sergey1011010) [reply storage/errors.go]: fixed

File: storage/sql_hooks.go (hunk 2 above)
Comment (Sergey1011010) [reply storage/sql_hooks.go]: fixed

File: storage/sql_hooks.go (hunk 3 above)
Comment (Sergey1011010) [reply storage/sql_hooks.go]: fixed

File: storage/storage.go (hunk 5 above)
Comment (Sergey1011010) [reply storage/storage.go]: Is there a standard? We can move it to `case postgres` and throw an error in default case

**Comment by Sergey1011010** [comment]
@bond95 they should be rewritten with mocked kafka to be unit tests

File: storage/storage.go (hunk 5 above)
Comment (Sergey1011010) [reply storage/storage.go]: done

**Comment by bond95** [comment]
> @bond95 they should be rewritten with mocked kafka to be unit tests

@Sergey1011010 but if this PR doesn't contain mocked kafka, maybe better to leave metrics and remove them when we will add mocked kafka?

**Comment by Sergey1011010** [comment]
@bond95 but then it will be integration tests, which we decided not to keep here :)

**Pull request merged**